alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -196,13 +201,15 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
 
         // Don't update monitor if this is a test monitor
         if (!isTempMonitor) {
-
-            // TODO: Check for race condition against the update monitor api
-            // This does the update at the end in case of errors and makes sure all the queries are executed
-            val updatedMonitor = monitor.copy(lastRunContext = updatedLastRunContext)
-            // note: update has to called in serial for shards of a given index.
-            // make sure this is just updated for the specific query or at the end of all the queries
-            updateMonitor(monitorCtx.client!!, monitorCtx.xContentRegistry!!, monitorCtx.settings!!, updatedMonitor)
+            updateMonitorMetadata(monitorCtx.client!!, monitorCtx.settings!!, monitorMetadata.copy(lastRunContext = updatedLastRunContext))
+            if (monitor.metadataId.isNullOrEmpty() || monitor.metadataId != monitorMetadata.id) {
+                updateMonitor(
+                    monitorCtx.client!!,
+                    monitorCtx.xContentRegistry!!,
+                    monitorCtx.settings!!,
+                    monitor.copy(metadataId = monitorMetadata.id)
+                )
+            }
```

Comment:
Why do we need to update the monitor now, since metadata is already updated with mutable information?

Reply:
We update the monitor if the metadata was not created for some reason or the metadata doc that this monitor was pointing to was never created initially, so a new metadata doc was created and this monitor needs to be updated to point to that metadata doc.

Reply:
This has been fixed by making the metadata id based on the monitor id so this is no longer needed